scripts/ioc_deploy.py

```diff
@@ -757,6 +833,74 @@ def _clone(
     return subprocess.run(cmd, **kwds)
 
 
+def _tag(
+    release: str,
+    message: str = "",
+    working_dir: str = "",
+    verbose: bool = False,
+) -> subprocess.CompletedProcess:
+    """
+    Create a tag on a github repo or raise a subprocess.CalledProcessError
+
+    Allow a release message but do not require it.
+    Avoid opening the editor dialog because this breaks the script.
+    """
+    cmd = ["git", "tag", release]
+    if message:
+        cmd.extend(["-a", "-m", message])
+    kwds = {"check": True}
+    if working_dir:
+        kwds["cwd"] = working_dir
+    if not verbose:
+        kwds["stdout"] = subprocess.PIPE
+        kwds["stderr"] = subprocess.PIPE
+    logger.debug(f"Calling '{' '.join(cmd)}' with kwargs {kwds}")
+    return subprocess.run(cmd, **kwds)
+
+
+def _push_tag(
+    release: str,
+    working_dir: str = "",
+    verbose: bool = False,
+) -> subprocess.CompletedProcess:
+    """
+    Push a tag to github or raise a subprocess.CalledProcessError
+
+    Relies on their being an existing git clone with origin set
+    appropriately and a local tag matching "release".
+    """
+    cmd = ["git", "push", "origin", release]
+    kwds = {"check": True}
+    if working_dir:
+        kwds["cwd"] = working_dir
+    if not verbose:
+        kwds["stdout"] = subprocess.PIPE
+        kwds["stderr"] = subprocess.PIPE
+    logger.debug(f"Calling '{' '.join(cmd)}' with kwargs {kwds}")
+    return subprocess.run(cmd, **kwds)
+
+
+def get_last_commit_info(
+    working_dir: str = "",
+    verbose: bool = False,
+) -> str:
+    """
+    Return the most recent commit message or raise a subprocess.CalledProcessError
+    """
+    cmd = ["git", "log", "-1"]
```

Comment:
```suggestion
    cmd = ["git", "log", "-1", "--pretty=%B"]
```
For me this gets all the nitty gritty from the log output as well, adding a few arguments can strip that

Reply:
This doesn't work for a shallow clone and I do not know why

Reply:
Weird, this works for me in a clone `--depth 1`, maybe I'm doing something differently?
<details><summary>Details</summary>
<p>

```bash
(base)roberttk@PC100147:~$ git clone --depth 1 git@github.com:tangkong/hello-world
Cloning into 'hello-world'...
remote: Enumerating objects: 3, done.
remote: Counting objects: 100% (3/3), done.
remote: Compressing objects: 100% (2/2), done.
remote: Total 3 (delta 0), reused 3 (delta 0), pack-reused 0 (from 0)
Receiving objects: 100% (3/3), done.
(base)roberttk@PC100147:~$ cd hello-world/
(base)roberttk@PC100147:~/hello-world(master -)$ git log
commit e2e299826a1f0d96bf1f2e5448d33032f85157f2 (grafted, HEAD -> master, tag: v0.0.1, origin/master, origin/HEAD)
Author: tangkong <35379409+tangkong@users.noreply.github.com>
Date:   Fri Jan 12 18:17:25 2018 +0000

    Merge pull request #1 from tangkong/readme-edits

    Update1 approved
(base)roberttk@PC100147:~/hello-world(master -)$ git log -1 --pretty=%B
Merge pull request #1 from tangkong/readme-edits

Update1 approved
```

</p>
</details>

Reply:
I swear this was failing for me last week but now I see the same

Reply:
I after seeing it fail (somehow) last week and switching to the full log, I ended up really appreciating seeing the full thing including the timestamp, user, etc., so it's more obvious if you are accidentally tagging prior to pushing or merging the PR.

Reply:
So to that end I'm going to change the docstring instead of the command

Reply:
The only thing a user might not care to see is the commit hash, but then again I personally wouldn't mind it.  Sounds good to me